clients/admin-ui/src/pages/management/domains.tsx

```diff
@@ -39,17 +38,78 @@ const CORSConfigurationPage: NextPage = () => {
   const { isLoading: isLoadingGetQuery } = useGetConfigurationSettingsQuery({
     api_set: true,
   });
-  const corsOrigins = useAppSelector(selectCORSOrigins());
+  const { isLoading: isLoadingConfigSetQuery } =
+    useGetConfigurationSettingsQuery({
+      api_set: false,
+    });
+  const currentSettings = useAppSelector(selectCORSOrigins);
+  const apiSettings = currentSettings.apiSet;
+  const configSettings = currentSettings.configSet;
   const applicationConfig = useAppSelector(selectApplicationConfig());
   const [putConfigSettingsTrigger, { isLoading: isLoadingPutMutation }] =
     usePutConfigurationSettingsMutation();
 
   const toast = useToast();
 
+  const isValidURL = (value: string | undefined) => {
+    if (!value) {
+      return false;
+    }
+    try {
+      const url = new URL(value);
+      return url.protocol === "https:" || url.protocol === "http:";
+    } catch (e) {
+      return false;
+    }
+    return true;
+  };
+
+  const containsNoWildcard = (value: string | undefined) => {
```

Comment:
Love how specific these validation rules are!

Reply:
👍 

A bit too many LoC for them and I do repeat some code a couple times, but it's because I wanted to test for errors and show unique messages for each

If I was being *really* good I'd push these into a shared `*-helpers.ts` function somewhere with unit tests, but alas I've already overengineered this one